app/common/state/contextMenuState.js

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const assert = require('assert')
+const eventUtil = require('../../../js/lib/eventUtil.js')
+const appActions = require('../../../js/actions/appActions.js')
+const windowAction = require('../../../js/actions/windowActions.js')
+const config = require('../../../js/constants/config.js')
+const CommonMenu = require('../commonMenu.js')
+const locale = require('../../../js/l10n.js')
+const { makeImmutable, isMap } = require('./immutableUtil')
+
+const validateAction = function (action) {
+  action = makeImmutable(action)
+  assert.ok(isMap(action), 'action must be an Immutable.Map')
+  return action
+}
+
+const validateState = function (state) {
+  state = makeImmutable(state)
+  assert.ok(isMap(state), 'state must be an Immutable.Map')
+  return state
+}
+
+const contextMenuState = {
+  setContextMenu: (state, action) => {
```

Comment:
the helpers are meant to be more general purpose so this should probably take `detail` as a parameter instead of `action`